core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -313,27 +252,7 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
     typedef Impl::ParallelReduce<FunctorType, TeamPolicy<Properties...>,
                                  ReducerType>
         closure_type;
-    typedef Impl::FunctorValueTraits<FunctorType, typename traits::work_tag>
-        functor_value_traits;
-
-    cudaFuncAttributes attr =
-        CudaParallelLaunch<closure_type, typename traits::launch_bounds>::
-            get_cuda_func_attributes();
-    const int block_size =
-        Kokkos::Impl::cuda_get_opt_block_size<FunctorType,
-                                              typename traits::launch_bounds>(
-            space().impl_internal_space_instance(), attr, f,
-            (size_t)vector_length(),
-            (size_t)team_scratch_size(0) + 2 * sizeof(double),
-            (size_t)thread_scratch_size(0) + sizeof(double) +
-                ((functor_value_traits::StaticValueSize != 0)
-                     ? 0
-                     : functor_value_traits::value_size(f)));
-    // Currently we require Power-of-2 team size for reductions.
-    int p2 = 1;
-    while (p2 <= block_size) p2 *= 2;
-    p2 /= 2;
-    return p2 / vector_length();
+    return team_size_recommeded_impl<closure_type>(f);
```

Comment:
probably this should be `recommended` with an n ...

Reply:
also the function later is named: `internal_team_size_recommended` but can we rename them both with "imp" as prefix instead of internal?